### PR TITLE
feat(cognito): user pools - sign in with apple

### PR DESCRIPTION
**Pull request by piotrmoszkowicz**
Added Sign In With Apple provider to `@aws-cdk/aws-cognito`. That's my first PR here, so bear with me, I hope I haven't made any mistakes, I've been following the docs carefully :)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-cognito/README.md [hunk 1]
```diff
@@ -418,6 +418,7 @@ The following third-party identity providers are currently supported in the CDK
 - [Login With Amazon](https://developer.amazon.com/apps-and-games/login-with-amazon)
 - [Facebook Login](https://developers.facebook.com/docs/facebook-login/)
 - [Google Login](https://developers.google.com/identity/sign-in/web/sign-in)
+- [Sing In With Apple](https://developer.apple.com/sign-in-with-apple/get-started/)
```
Comment (nija-at) [inline packages/@aws-cdk/aws-cognito/README.md:421]:
```suggestion
- [Sign In With Apple](https://developer.apple.com/sign-in-with-apple/get-started/)
```

File: packages/@aws-cdk/aws-cognito/lib/user-pool-idps/apple.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+   * The client id recognized by Apple APIs.
+   * @see https://developer.apple.com/documentation/sign_in_with_apple/clientconfigi/3230948-clientid
+   */
+  readonly clientId: string;
+  /**
+   * The teamId for Apple APIs to authenticate the client.
+   */
+  readonly teamId: string;
+  /**
+   * The keyId (which has to be later supplied as `privateKey`) for Apple APIs to authenticate the client.
```
Comment (nija-at) [inline packages/@aws-cdk/aws-cognito/lib/user-pool-idps/apple.ts:20]: Does this mean keyid and private key are always the same?

**Review by nija-at — CHANGES_REQUESTED** [review]
Thanks for taking the time and submitting this PR 😊 
See my comments below

Could you also add an integ test similar to this - https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-cognito/test/integ.user-pool-idp.amazon.ts ?
Take a look at the stack verification steps on that test. Add ones similar to that to the test.

File: packages/@aws-cdk/aws-cognito/lib/user-pool-idps/apple.ts (hunk 2 above)
Comment (piotrmoszkowicz) [reply packages/@aws-cdk/aws-cognito/lib/user-pool-idps/apple.ts:20]: No it's not, but you are right, the comment is terrible. I will work on it :)

**Comment by piotrmoszkowicz** [comment]
@nija-at Adjusted the code according to your instructions ;)

**Review by nija-at — APPROVED** [review]

**Comment by piotrmoszkowicz** [comment]
@nija-at Test hasn't passed due change in default retention policy for `aws-cognito`. Therefor the `json` wasn't up-to-date. I've updated it, tests shall pass now 🤞

**Review by nija-at — APPROVED** [review]

**Pull request merged**